CMakeLists.txt

```diff
@@ -29,83 +37,129 @@ message( STATUS "-------- cpp2py version and git hash detection -------------")
 
 find_package(Git)
 
-# Get the Git Hash
+# get the Git Hash
 execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
- WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
- OUTPUT_VARIABLE   CPP2PY_GIT_HASH
- OUTPUT_STRIP_TRAILING_WHITESPACE
+  WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
+  OUTPUT_VARIABLE   CPP2PY_GIT_HASH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
 )
 
 message(STATUS "Cpp2py version : ${CPP2PY_VERSION}")
 message(STATUS "Git hash : ${CPP2PY_GIT_HASH}")
 
-# LIBCLANG : try to set LIBCLANG_LOCATION and LIBCLANG_CXX_FLAGS automatically
-MESSAGE(STATUS "-------- LibClang detection -------------")
+#--------------------------------------------------------
+#   Libclang
+#--------------------------------------------------------
+message(STATUS "-------- LibClang detection -------------")
 
 find_package(LibClang)
 
-MESSAGE(STATUS "LibClang location: ${LIBCLANG_LOCATION}")
-MESSAGE(STATUS "LibClang additional flags: ${LIBCLANG_CXX_FLAGS}")
+message(STATUS "LibClang location: ${LIBCLANG_LOCATION}")
+message(STATUS "LibClang additional flags: ${LIBCLANG_CXX_FLAGS}")
 
 # allow user to modify
 set(LIBCLANG_CXX_FLAGS "${LIBCLANG_CXX_FLAGS}" CACHE STRING "Additional flags to be passed to libclang when parsing with clang")
 set(LIBCLANG_LOCATION "${LIBCLANG_LOCATION}" CACHE STRING "Location of the libclang library")
 
-# --- Python ---
+#--------------------------------------------------------
+#   Python and Numpy
+#--------------------------------------------------------
+find_package(Python REQUIRED COMPONENTS Interpreter Development NumPy)
+
+# check required python packages
+function(exec_python_script the_script output_var_name)
+  execute_process(COMMAND ${Python_EXECUTABLE} -c "${the_script}"
+    OUTPUT_VARIABLE res RESULT_VARIABLE returncode OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(NOT returncode EQUAL 0)
+    message(FATAL_ERROR "The script : ${the_script} \n did not run properly in the Python interpreter. Check your python installation.")
+  endif()
+  set(${output_var_name} ${res} PARENT_SCOPE)
+endfunction()
+
+exec_python_script("import mako.template" nulle)
+exec_python_script("import numpy" nulle)
+exec_python_script("import scipy" nulle)
+if(Build_Documentation)
+  exec_python_script("import clang.cindex" nulle)
+endif()
+
+# set variables used in the current and in dependent projects
+set(PYTHON_EXECUTABLE ${Python_EXECUTABLE} CACHE STRING "Python executable")
+if(Python_SOABI)
+  set(PYTHON_MODULE_EXT .${Python_SOABI}.so CACHE STRING "Python module extension")
+else()
+  set(PYTHON_MODULE_EXT .so CACHE STRING "Python module extension")
+endif()
+if(BUILD_DEBIAN_PACKAGE)
+  set(PYTHON_LIB_DEST_DIR dist-packages)
+else()
+  set(PYTHON_LIB_DEST_DIR site-packages)
+endif()
+set(PYTHON_LIB_DEST_ROOT lib/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/${PYTHON_LIB_DEST_DIR} CACHE PATH "Python module directory")
```

Comment:
```suggestion
string(REGEX REPLACE "/.*/lib" "lib" PYTHON_LIB_DEST_ROOT ${Python_SITEARCH})
set(PYTHON_LIB_DEST_ROOT ${PYTHON_LIB_DEST_ROOT} CACHE PATH "Python library destination")
```

Reply:
Does this still work when we want to build debian packages?

Reply:
I believe this should yield the correct path containing `dist-packages` instead of `site-packages` on Ubuntu.
Could you please check?

Reply:
Ah yes, you are right! That should work.